primitives/src/runtimes/mainnet.rs

```diff
@@ -0,0 +1,191 @@
+//! Darwinia Runtime
+#![cfg(feature = "runtime")]
+
+use crate::{
+	chain::{
+		ethereum::EthereumRelayHeaderParcel, proxy_type::ProxyType, RelayAffirmation,
+		RelayAffirmationId, RelayVotingState,
+	},
+	frame::{
+		bridge::relay_authorities::EthereumRelayAuthorities,
+		ethereum::{
+			backing::EthereumBacking, game::EthereumRelayerGame, issuing::EthereumIssuing,
+			relay::EthereumRelay,
+		},
+		proxy::Proxy,
+		sudo::Sudo,
+		technical_committee::TechnicalCommittee,
+	},
+};
+
+use crate::frame::bridge::relay_authorities::RelayAuthority;
+use substrate_subxt::{
+	balances::{AccountData, Balances},
+	extrinsic::DefaultExtra,
+	register_default_type_sizes, sp_core,
+	sp_runtime::{
+		generic::Header,
+		traits::{BlakeTwo256, IdentifyAccount, Verify},
+		MultiAddress, MultiSignature, OpaqueExtrinsic,
+	},
+	system::System,
+	EventTypeRegistry, Runtime,
+};
+
+use sp_core::{H160, H256, U256};
+use codec::{Decode, Encode};
+use crate::chain::ethereum::{EcdsaAddress, EcdsaSignature, EcdsaMessage};
+
+type SessionIndex = u32;
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct Log {
+	address: H160,
+	topics: Vec<H256>,
+	data: Vec<u8>,
+}
+
+/// Darwinia Runtime
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct MainnetRuntime;
```

Comment:
Since we have two mainnets Darwinia and Crab, I suggest to use Darwinia directly here.

Reply:
for now, I use mainnet, crab, pangolin as chain names.

The crate's name is `darwinia`. if we use `DarwiniaRuntime`, is it better to change the crate's name to a new one like `chain`?

Reply:
We should use the runtime spec name here, crab is not a darwinia chain.

Though the crab runtime is located at darwinia repo, we follow the way of polkadot repo, for the crate names, how about runtime-client-types, or runtime-types.

Non darwinia runtime could also locate here, e.g. polkadot or kusama or some other substrate runtime.